File: typescript-connector/src/models/ProductBatch.ts
// Class from DFC Business Ontology: #ProductBatch
import { SemanticObject } from "../core/SemanticObject.js";
import { WhatSubject, type WhatSubjectParams } from "./WhatSubject.js";

export interface ProductBatchParams extends WhatSubjectParams {
  batchNumber?: string;
  bestBeforeDate?: string;
  expiryDate?: string;
  productionDate?: string;
  identifies?: string[];
  traces?: string[];
}
export class ProductBatch extends WhatSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:ProductBatch";
  }

  batchNumber?: string;
  bestBeforeDate?: string;
  expiryDate?: string;
  productionDate?: string;
  identifies?: string[];
  traces?: string[];

  constructor(
    semanticId: string,
    params?: ProductBatchParams,
  ) {
    super(semanticId, params);
    this.batchNumber = params?.batchNumber;
    this.bestBeforeDate = params?.bestBeforeDate;
    this.expiryDate = params?.expiryDate;
    this.productionDate = params?.productionDate;
    this.identifies = params?.identifies;
    this.traces = params?.traces;
    this.semanticType = ProductBatch.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:batchNumber", () => this.batchNumber);
    this.registerSemanticProperty("dfc-b:bestBeforeDate", () => this.bestBeforeDate);
    this.registerSemanticProperty("dfc-b:expiryDate", () => this.expiryDate);
    this.registerSemanticProperty("dfc-b:productionDate", () => this.productionDate);
    this.registerSemanticProperty("dfc-b:identifies", () => this.identifies);
    this.registerSemanticProperty("dfc-b:traces", () => this.traces);
  }
  static {
    SemanticObject.typeRegistry.set(ProductBatch.SEMANTIC_TYPE, ProductBatch);
  }
}

File: ruby-gem/lib/models/product_batch.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #ProductBatch
require_relative '../core/semantic_object'
require_relative 'what_subject'


module DfcLinkmlConnector
  module Models
    class ProductBatch < WhatSubject
      SEMANTIC_TYPE = "dfc-b:ProductBatch".freeze

      # @return [String]
      attr_accessor :batch_number

      # @return [String]
      attr_accessor :best_before_date

      # @return [String]
      attr_accessor :expiry_date

      # @return [String]
      attr_accessor :production_date

      # @return [String]
      attr_accessor :identifies

      # @return [String]
      attr_accessor :traces

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, batchNumber: nil, bestBeforeDate: nil, expiryDate: nil, productionDate: nil, identifies: nil, traces: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, batchNumber: nil, bestBeforeDate: nil, expiryDate: nil, productionDate: nil, identifies: nil, traces: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @batch_number = batchNumber
        @best_before_date = bestBeforeDate
        @expiry_date = expiryDate
        @production_date = productionDate
        @identifies = identifies
        @traces = traces
        self.semanticType = "dfc-b:ProductBatch"
        registerSemanticProperty("dfc-b:ProductBatch:batch_number", &method("batch_number")).valueSetter = method("batch_number=")
        registerSemanticProperty("dfc-b:ProductBatch:best_before_date", &method("best_before_date")).valueSetter = method("best_before_date=")
        registerSemanticProperty("dfc-b:ProductBatch:expiry_date", &method("expiry_date")).valueSetter = method("expiry_date=")
        registerSemanticProperty("dfc-b:ProductBatch:production_date", &method("production_date")).valueSetter = method("production_date=")
        registerSemanticProperty("dfc-b:ProductBatch:identifies", &method("identifies")).valueSetter = method("identifies=")
        registerSemanticProperty("dfc-b:ProductBatch:traces", &method("traces")).valueSetter = method("traces=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
